### PR TITLE
Add general fallback for rem2pi

### LINKED ISSUE #44715: `rem2pi(x::Integer, r::RoundingMode)` is missing.
Currently we only defined `rem2pi(x::Int32, r::RoundingMode)`, `rem2pi(x::Int64, r::RoundingMode)`.
A general fallback seems make sense.
```julia
function rem2pi(x::Integer, r::RoundingMode)
    fx = float(x)
    fx == x || throw(ArgumentError("$(typeof(x)) argument to rem2pi is too large: $x"))
    rem2pi(fx, r)
end
```

### PR DESCRIPTION
**Pull request by 11happy**
**Overview:**
- This PR fixes #44715 by adding a general fallback.

**Testing:**
- Tested the updated code.
- Verified that other functionalities remain unaffected.

**Dependencies:**
- No dependencies on other pull requests.

**CC:**
- @N5N3

### CODE REVIEW

**Comment by barucden** [comment]
Are the other methods (for `Int32` and `Int64`) any better than this fallback? I see that the one for `Int32` omits the condition, but the one for `Int64` seems exactly like the newly-added, general method.

**Comment by 11happy** [comment]
> Are the other methods (for `Int32` and `Int64`) any better than this fallback? I see that the one for `Int32` omits the condition, but the one for `Int64` seems exactly like the newly-added, general method.

like there's not much significant difference, Int32 just omits the condition.

File: base/math.jl [hunk 1]
```diff
@@ -1474,6 +1474,12 @@ function rem2pi(x::Int64, r::RoundingMode)
     fx == x || throw(ArgumentError("Int64 argument to rem2pi is too large: $x"))
     rem2pi(fx, r)
 end
+# general fallback
+function rem2pi(x::Integer, r::RoundingMode)
+    fx = float(x)
+    fx == x || throw(ArgumentError("$(typeof(x)) argument to rem2pi is too large: $x"))
```
Comment (N5N3) [inline base/math.jl:1480]:
```suggestion
    fx == x || throw(ArgumentError(LazyString(typeof(x), " argument to rem2pi is too large: ", x)))
```
LGTM, and the `Int64` fallback could be removed.

**Comment by barucden** [comment]
The method for `Int32` avoids the check `x == Float64(x)` likely because `Int32` is always representable as `Float64` exactly. The same holds even for `Int8` and `Int16`, right? We might extend that method to accept `Union{Int8, Int16, Int32}`.

**Comment by ViralBShah** [comment]
cc @oscardssmith

**Comment by 11happy** [comment]
> The method for `Int32` avoids the check `x == Float64(x)` likely because `Int32` is always representable as `Float64` exactly. The same holds even for `Int8` and `Int16`, right? We might extend that method to accept `Union{Int8, Int16, Int32}`.

sure could do that , should I?

**Review by oscardssmith — APPROVED** [review]

**Comment by N5N3** [comment]
> sure could do that , should I?

I guess this is not a must. LLVM is clever enough to eliminate the error path:
```julia
julia> function myrem2pi(x::Integer, r::RoundingMode)                                          
           fx = float(x)                                                                       
               fx == x || throw(ArgumentError(LazyString(typeof(x), " argument to rem2pi is too large: ", x)))                                                                                
           rem2pi(fx, r)                                                                       
       end
myrem2pi (generic function with 1 method)

julia> @code_llvm myrem2pi(Int32(1), RoundUp)
; Function Signature: myrem2pi(Int32, Base.Rounding.RoundingMode{:Up})
;  @ REPL[1]:1 within `myrem2pi`
; Function Attrs: uwtable
define double @julia_myrem2pi_6707(i32 signext %"x::Int32") #0 {
L5:
;  @ REPL[1]:2 within `myrem2pi`
; ┌ @ float.jl:356 within `float`
; │┌ @ float.jl:329 within `AbstractFloat`
; ││┌ @ float.jl:221 within `Float64`
     %0 = sitofp i32 %"x::Int32" to double
; └└└
;  @ REPL[1]:4 within `myrem2pi`
  %1 = call double @j_rem2pi_6713(double %0)
  ret double %1
}
```

**Comment by barucden** [comment]
What about removing the `Int32` method then, as you suggested @N5N3?

**Comment by N5N3** [comment]
@barucden That make sense to me. But perhaps we need some test to ensure there's no regression.

**Pull request merged**